django_stomp/execution.py

```diff
@@ -47,7 +48,7 @@ def main_logic() -> Optional[Listener]:
             logger.info("Starting listener...")
 
             def _callback(payload: Payload) -> None:
-                local_threading.request_id = payload.headers["correlation-id"] if "correlation-id" in payload.headers else None
+                local_threading.request_id = _get_or_create_correlation_id(payload)
```

Comment:
Inside `_get_or_create_correlation_id` only headers is used, no need for the entire payload that suggests that this method is using something beyond headers, which is not the case.